File: back_end/src/main/java/com/example/back_end/service/song/impl/SongService.java
package com.example.back_end.service.song.impl;

import com.example.back_end.model.song.Song;
import com.example.back_end.model.song.SongProjection;
import com.example.back_end.repository.song.ISongRepository;
import com.example.back_end.service.song.ISongService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class SongService implements ISongService {
    @Autowired
    private ISongRepository songRepository;

    @Override
    public List<Song> getListSongBySearch(String searchKeyWord) {
        return songRepository.getListSongBySearch("%" + searchKeyWord + "%");
    }

    @Override
    public Page<SongProjection> getListSongBySearched(Pageable pageable, String searchKeyWord) {
        return songRepository.getListSongBySearched(pageable, "%" + searchKeyWord + "%");
    }

    @Override
    public SongProjection getSongBySongId(Long songId) {
        return songRepository.getSongBySongId(songId);
    }


}

File: font_end/src/services/SongService.js
import axios from "axios";


export async function getListSongBySearch(searchKeyWord) {
        try {
                const result = await axios.get(`http://localhost:8080/api/song?searchKeyWord=${searchKeyWord}`);
                return result.data;
        } catch (e) {
                console.log(e);
        }

}
export async function searchSong(searchKeyWord, pageNumber) {
        try {
                const result = await axios.get(`http://localhost:8080/api/song/search?searchKeyWord=${searchKeyWord}&page=${pageNumber}&size=1`);
                return result.data;
        } catch (e) {
                console.log(e);
        }

}
export async function getSongById(songId) {
        try {
                const result = await axios.get(`http://localhost:8080/api/song/detail/${songId}`);
                return result.data;
        } catch (e) {
                console.log(e);
        }

}